### ISSUE TITLE
[IDISP011] False positive, if variable is disposed and reassigned on only one execution path

### ISSUE THREAD

**Issue by Pretasoc** [DotNetAnalyzers/IDisposableAnalyzers#239]
Consider the following example:

``` csharp
class Foo : IDisposable { /* Implementation omited */ }
class Bar : IDisposable { /* Implementation omited */ }

IDisposable Foo()
{
    IDisposable value = new Foo();

    if(DoStuff())
    {
        value.Dispose();
        value = new Bar()
    }
   return value;
}
```

In this case a `IDISP011` is triggered at the return statement. Hoever not in this example

``` csharp
class Foo : IDisposable { /* Implementation omited */ }
class Bar : IDisposable { /* Implementation omited */ }

IDisposable Foo()
{
    IDisposable value = new Foo();

    value.Dispose();
    value = new Bar()

   return value;
}
```

**Comment by JohanLarsson** [comment]
Yes, we can figure that out, thanks for reporting!

**Comment by JohanLarsson** [comment]
https://www.nuget.org/packages/IDisposableAnalyzers/3.4.5
